### PR TITLE
versatile-data-kit: Adding support to read SQM statement as DataFrame

### PR DESCRIPTION
**Pull request by victor-partico**
#Why
execute_query does not return column name, DataFrames are move useful

#What
Adding new method to job_input

#How has this been tested?
None, could not make build.sh run

#Type
New feature (non-breaking change which adds functionality)

Signed-off-by: Victor Mishovski <vmishovski@vmware.com>

### CODE REVIEW

**Comment by tozka** [comment]
Thanks for the proposal. I do think it's an interesting and important feature to be able to work with Pandas  easily. 

There are a few things though: 

* Маке sure to follow [vmwclabot](https://github.com/vmwclabot) Instructions 

* I recommend having an associated Github issue that explain what and why do we need this . The very least the "Why" section needs to be much more than 1 sentence . See https://github.com/vmware/versatile-data-kit/blob/main/CONTRIBUTING.md#how-to-prepare-your-change

* I am not sure the change should be part of "vdk-core" itself. It introduces a new dependency in vdk-core - "pandas" - and as vdk-core is used in every single plugin and everywhere (vdk related) it needs to have as little dependencies as possible to avoid dependencies conflicts or some variant of [dependency hell](https://en.wikipedia.org/wiki/Dependency_hell). 

You can consider making it a [plugin](https://github.com/vmware/versatile-data-kit/tree/main/projects/vdk-plugins) (vdk-pandas). Changes in vdk-core are much harder to be accepted as they have much bigger impact.
While we welcome all kinds of plugins. 

* Changes in vdk-core and in plugins must be in separate PRs - first for the change in vdk-core and then for the plugins. Though the above point may make this one moot.

**Comment by tozka** [comment]
Btw the reason build.sh and tests fail (at least what I can see from https://gitlab.com/vmware-analytics/versatile-data-kit/-/jobs/3274269747) is because "ModuleNotFoundError: No module named 'pandas'" 

New dependencies are added in setup.cfg (or setup.py) file in[ install_requires section](https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/#id5)

File: projects/vdk-core/src/vdk/api/job_input.py [hunk 1]
```diff
@@ -79,6 +81,29 @@ def execute_query(self, query_as_utf8_string) -> List[List]:
...
+        arguments (get_arguments()) will take precedence over properties (get_property()).
+
+        Note:
+            Parameters are case sensitive.
+            If a query parameter is not present in Data Job properties or arguments it will not be replaced in the query
+            and most likely result in query failure.
+
+        Example usage:
+            job_input.set_all_properties({'target_table': 'history.people','etl_run_date_column': 'pa__arrival_ts'})
+            query_result = job_input.execute_query("SELECT * FROM {target_table} WHERE {etl_run_date_column} > now() - interval 2 hours")
```
Comment (ivakoleva) [inline projects/vdk-core/src/vdk/api/job_input.py:102]: include `read_sql_as_dataframe` to the Example usage;
also include into plugin's README.md added

**Review by ivakoleva — CHANGES_REQUESTED** [review]

File: projects/vdk-core/src/vdk/internal/builtin_plugins/connection/managed_connection_base.py [hunk 2]
```diff
@@ -157,6 +158,31 @@ def execute_query(self, query: str) -> List[List[Any]]:
                 List[List[Any]], res
             )  # we return None in case of DML. This is not PEP249 compliant, but is more convenient
 
+    def read_sql_as_dataframe(self, query: str) -> pd.DataFrame:
+        """Read SQL statement and returns the result as pandas DataFrame"""
+        with self.connect() as db:
+            try:
+                res = pd.read_sql(query, con=db)
```
Comment (murphp15) [inline projects/vdk-core/src/vdk/internal/builtin_plugins/connection/managed_connection_base.py:165]: I would probably just do `return pd.read_sql(query, con=db)` here and then `return None` at the end of the catch statement. But might just be personal preference.

File: projects/vdk-plugins/vdk-postgres/src/vdk/plugin/postgres/postgres_connection.py [hunk 3]
```diff
@@ -112,3 +113,9 @@ def execute_query(self, query: str) -> List[List[Any]]:
             return super().execute_query(query)
         finally:
             self.commit()
+
+    def read_sql_as_dataframe(self, query: str) -> List[List[Any]]:
```
Comment (murphp15) [inline projects/vdk-plugins/vdk-postgres/src/vdk/plugin/postgres/postgres_connection.py:117]: Should the return type not be dataframe?

**Comment by tozka** [comment]
We talked with Victor and decided we will go with documentation improvement since ultimately this is currently possible with VDK using 
```
pandas.read_sql("select * from x" , job_input.get_managed_connection()) 
```

**Pull request closed without merging**